lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -69,7 +69,62 @@ The following arguments were given to MyLibrary.foo/1:
 
 ## Alternative return types
 
-TODO
+#### Problem
+
+This anti-pattern refers to functions that receive options (for example, *keyword list*) parameters that drastically change their return type. Because options are optional and sometimes set dynamically, if they change the return type it may be hard to understand what the function actually returns.
+
+#### Example
+
+An example of this anti-pattern, as shown below, is when a library (for example, `AlternativeInteger`) has a multi-clause function `parse/2` with many alternative return types. Depending on the options received as a parameter, the function will have a different return type.
+
+```elixir
+defmodule AlternativeInteger do
+  def parse(string, opts) when is_list(opts) do
+    case opts[:discard_rest] do
+      true -> String.to_integer(string)
+      _    -> Integer.parse(string)
+    end
+  end
+
+  def parse(string, opts \\ :default) do
+    case opts do
+      :default -> Integer.parse(string)
+    end
+  end
```

Comment:
I think the `parse/2` function shows this issue by itself. I don't think this clause adds anything, and it shows a sort-of-unconventional use of default options (usually you'd have `opts \\ []` and then `Keyword.get(opts, :discard_rest, false)`).

```suggestion
```